script/EntryEditor.js

```diff
@@ -14,28 +14,39 @@ var EntryEditor = function($form) {
     $form.find('.struct .hashint').tooltip();
 
     /**
-     * Attach datepicker to date types
+     * Attach datepicker to date types.
+     * Only if browser does not support HTML5 date input.
      */
-    $form.find('input.struct_date').datepicker({
-        dateFormat: 'yy-mm-dd'
-    });
+    var testelem = document.createElement('input');
+    testelem.setAttribute("type", "date");
```

Comment:
Hmm, you're setting the type in the HTML of the actual input field - I wonder what `attr()` does return on browsers not supporting it. I would assume that they return `text` as well. If that's the case the check could be done on the real element without the need to create a new element.

Reply:
jquery's `attr()` returns what is written in the HTML code, that is "date", whereas the DOM element `.type` gives "text". Thus, I did not know how to filter this with jquery's `find()`. But '.filter()' should work and I shall simplify the code accordingly.

Reply:
very elegant solution!